app/interactors/update_user.rb

```diff
@@ -2,8 +2,23 @@ class UpdateUser
   include Interactor
 
   def call
-    user = context.current_user.update(context.permitted_params)
+    @params = context.permitted_params
+    @user = context.current_user
 
-    context.fail!(errors: user.errors) unless user
+    update_address
+    @user.update_without_password(@params) if @params[:email]
+    if @params[:current_password]
+      context.fail! unless @user.update_with_password(@params)
+    end
+  rescue
+    context.fail!
+  end
+
+  private
+
+  def update_address
```

Comment:
Can we update addresses via user using nested attributes?

Reply:
No. It throws exception "type mismatch"